modules/csg/csg_shape.cpp

```diff
@@ -943,6 +901,23 @@ Array CSGShape3D::get_meshes() const {
 	return Array();
 }
 
+PackedStringArray CSGShape3D::get_configuration_warnings() const {
+	PackedStringArray warnings = Node::get_configuration_warnings();
+	const String manifold_error_message = "The CSGShape3D has an empty shape.";
+	const String manifold_info_message = "CSGShape3D empty shapes typically occur because the mesh is not manifold. "
+										 "A manifold mesh forms a solid object without gaps, holes, or loose edges. Each edge must be a member of exactly two faces.";
+	const CSGShape3D *current_shape = this;
+	while (current_shape) {
+		if (!current_shape->brush || current_shape->brush->faces.is_empty()) {
+			warnings.push_back(manifold_error_message);
+			warnings.push_back(manifold_info_message);
```

Comment:
Declare a single warning (otherwise, it's counted as two separate warnings by the editor) and make it localizable:

```suggestion
			warnings.push_back(RTR("The CSGShape3D has an empty shape.\nCSGShape3D empty shapes typically occur because the mesh is not manifold.\nA manifold mesh forms a solid object without gaps, holes, or loose edges.\nEach edge must be a member of exactly two faces."));
```

Reply:
> We should fix this in the dialog display instead.

I'm not sure if this is a bug or a feature enhancement. Can you write it up?